internal/bridge/client.go

```diff
@@ -45,6 +46,172 @@ type Client struct {
+	DiskUsage              *ClusterDiskUsageApiResource `json:"disk_usage,omitempty"`
+	Environment            string                       `json:"environment,omitempty"`
+	Host                   string                       `json:"host,omitempty"`
+	IsHA                   bool                         `json:"is_ha,omitempty"`
+	IsProtected            bool                         `json:"is_protected,omitempty"`
+	IsSuspended            bool                         `json:"is_suspended,omitempty"`
+	Keychain               string                       `json:"keychain_id,omitempty"`
+	MaintenanceWindowStart int64                        `json:"maintenance_window_start,omitempty"`
+	MajorVersion           int                          `json:"major_version,omitempty"`
+	Memory                 float64                      `json:"memory,omitempty"`
+	ClusterName            string                       `json:"name,omitempty"`
+	Network                string                       `json:"network_id,omitempty"`
+	Parent                 string                       `json:"parent_id,omitempty"`
+	Plan                   string                       `json:"plan_id,omitempty"`
+	PostgresVersion        intstr.IntOrString           `json:"postgres_version_id,omitempty"`
+	Provider               string                       `json:"provider_id,omitempty"`
+	Region                 string                       `json:"region_id,omitempty"`
+	Replicas               []*ClusterApiResource        `json:"replicas,omitempty"`
+	Storage                int64                        `json:"storage,omitempty"`
+	Tailscale              bool                         `json:"tailscale_active,omitempty"`
+	Team                   string                       `json:"team_id,omitempty"`
+	LastUpdate             string                       `json:"updated_at,omitempty"`
+	ResponsePayload        v1beta1.SchemalessObject     `json:""`
+}
+
+func (c *ClusterApiResource) AddDataToClusterStatus(cluster *v1beta1.CrunchyBridgeCluster) {
+	cluster.Status.ClusterName = c.ClusterName
+	cluster.Status.Host = c.Host
+	cluster.Status.ID = c.ID
+	cluster.Status.IsHA = c.IsHA
+	cluster.Status.IsProtected = c.IsProtected
+	cluster.Status.MajorVersion = c.MajorVersion
+	cluster.Status.Plan = c.Plan
+	cluster.Status.Storage = c.Storage
+	cluster.Status.Responses.Cluster = c.ResponsePayload
+}
+
+type ClusterList struct {
+	Clusters []*ClusterApiResource `json:"clusters"`
+}
+
+// ClusterDiskUsageApiResource hold information on disk usage for a particular cluster.
+type ClusterDiskUsageApiResource struct {
+	DiskAvailableMB int64 `json:"disk_available_mb,omitempty"`
+	DiskTotalSizeMB int64 `json:"disk_total_size_mb,omitempty"`
+	DiskUsedMB      int64 `json:"disk_used_mb,omitempty"`
+}
+
+// ClusterGroupApiResource holds information on a ClusterGroup
+type ClusterGroupApiResource struct {
+	ID       string                `json:"id,omitempty"`
+	Clusters []*ClusterApiResource `json:"clusters,omitempty"`
```

Comment:
`ClusterApiResource.ClusterGroup.Clusters` == `ClusterApiResource`, yeah? Just without any clustergroup info of their own? (I never hit this part of the API)

Reply:
A slice of ClusterApiResource, yeah.